pkg/build/controller/controller.go

```diff
@@ -216,15 +216,24 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 	case kapi.PodRunning:
 		// The pod's still running
 		nextStatus = buildapi.BuildPhaseRunning
-	case kapi.PodSucceeded, kapi.PodFailed:
+	case kapi.PodSucceeded:
 		// Check the exit codes of all the containers in the pod
 		nextStatus = buildapi.BuildPhaseComplete
-		for _, info := range pod.Status.ContainerStatuses {
-			if info.State.Terminated != nil && info.State.Terminated.ExitCode != 0 {
-				nextStatus = buildapi.BuildPhaseFailed
-				break
+		if len(pod.Status.ContainerStatuses) == 0 {
```

Comment:
This was historic - in v1beta1 we didn't have PodFailed or Succeeded, and
we had to set the field to something, so we couldn't rely on PodFailed.
It's safe to fix it the correct way.

On Wed, Jul 29, 2015 at 11:19 AM, Ben Parees <notifications@github.com>
wrote:

> In pkg/build/controller/controller.go
> <https://github.com/openshift/origin/pull/3936#discussion_r35771914>:
>
> >  		// Check the exit codes of all the containers in the pod
> >  		nextStatus = buildapi.BuildPhaseComplete
> > -		for _, info := range pod.Status.ContainerStatuses {
> > -			if info.State.Terminated != nil && info.State.Terminated.ExitCode != 0 {
> > -				nextStatus = buildapi.BuildPhaseFailed
> > -				break
> > +		if len(pod.Status.ContainerStatuses) == 0 {
>
> it does and it did, which is why i also added the check for kapi.PodFailed
> explicitly. previously (for unknown reasons) we treated kapi.PodFailed as a
> possible success.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/3936/files#r35771914>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift